File: backend/models/vehicle.py
from sqlalchemy import Column, Integer, String, DateTime
from sqlalchemy.orm import relationship
from datetime import datetime
from database import Base


class Vehicle(Base):
    __tablename__ = "vehicles"

    id = Column(Integer, primary_key=True, index=True)
    vehicle_id = Column(String(50), unique=True, index=True, nullable=False)
    license_plate = Column(String(20))
    vehicle_type = Column(String(50))
    status = Column(String(20), default="active")
    created_at = Column(DateTime, default=datetime.utcnow)

    locations = relationship(
        "GPSLocation",
        back_populates="vehicle",
        cascade="all, delete-orphan"
    )

File: frontend/Smart_Transport_System/lib/models/vehicle.dart
class Vehicle {
  final int id;
  final String vehicleId;
  final String? licensePlate;
  final String? vehicleType;
  final String status;

  Vehicle({
    required this.id,
    required this.vehicleId,
    this.licensePlate,
    this.vehicleType,
    required this.status,
  });

  factory Vehicle.fromJson(Map<String, dynamic> json) {
    return Vehicle(
      id: json["id"],
      vehicleId: json["vehicle_id"],
      licensePlate: json["license_plate"],
      vehicleType: json["vehicle_type"],
      status: json["status"],
    );
  }
}
